api/mod/src/routers/login.py

```diff
@@ -14,13 +17,19 @@
     "/",
     status_code=204,
     responses={
+        400: {"description": "No user found with that username"},
         401: {"description": "Failed to validate login"},
     },
     operation_id="login",
 )
 async def login(
     login_data: LoginData,
+    db: Session = Depends(get_db),
```

Comment:
Does this `Depends` thing ensure that we always use the same database connection? Or what else is this meant to do?

Reply:
For most requests we will need access to the database. And this "`Depends`" provides that database connection for the route because the route _depends_ on the database. For each request it will create a new session (as fat as I understood it). I dont know the full power of this mechanism yet, but here are the docs https://fastapi.tiangolo.com/tutorial/dependencies/
So i guess in this case its just a helper to ensure the `get_db` function is invoked everywhere where we need it